src/hooks/usePlaceOrder.ts

```diff
@@ -0,0 +1,76 @@
+import BN from 'bn.js'
+import { toast } from 'react-toastify'
+
+import { TokenDetails, PlaceOrderParams as ExchangeApiPlaceOrderParams } from 'types'
+import { useState } from 'react'
+import { exchangeApi } from 'api'
+import { log } from 'utils'
+import { txOptionalParams } from 'utils/transaction'
+import { useWalletConnection } from './useWalletConnection'
+import { DEFAULT_ORDER_DURATION } from 'const'
+
+interface Params {
+  callback: () => void
+}
+
+interface PlaceOrderParams {
+  buyAmount: BN
+  buyToken: TokenDetails
+  sellAmount: BN
+  sellToken: TokenDetails
+}
+
+interface Result {
+  placeOrder: (params: PlaceOrderParams) => Promise<void>
+  isSubmitting: boolean
+}
+
+export const usePlaceOrder = ({ callback }: Params): Result => {
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const { userAddress } = useWalletConnection()
+
+  async function placeOrder({ buyAmount, buyToken, sellAmount, sellToken }: PlaceOrderParams): Promise<void> {
```

Comment:
Better memoize on `userAddress` as `placeOrder` can potentially be used as a prop, etc. down the line

Reply:
Done